src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/DateStatement.scala

```diff
@@ -22,8 +29,29 @@ private[querygeneration] object DateStatement {
 
     Option(expr match {
       case DateAdd(startDate, days) =>
-        ConstantString("DATEADD(day,") + convertStatement(days, fields) + "," +
-          convertStatement(startDate, fields) + ")"
+        ConstantString(SNOWFLAKE_DATEADD) +
+          blockStatement(
+            ConstantString("day,") +
+              convertStatement(days, fields) + "," +
+              convertStatement(startDate, fields)
+          )
+
+      // Snowflake has no direct DateSub function,
+      // it is pushdown by DATEADD with negative days
+      case DateSub(startDate, days) =>
+        ConstantString(SNOWFLAKE_DATEADD) +
+          blockStatement(
+            ConstantString("day, (0 - (") +
+              convertStatement(days, fields) + ") )," +
+              convertStatement(startDate, fields)
+          )
+
+      // AddMonths can't be pushdown to snowflake because their functionality is different.
+      // For Snowflake and Spark 2.3/2.4, AddMonths() will preserve the end-of-month information.
+      // But, Spark 3.0, it doesn't. For example,
+      // On spark 2.3/2.4, "2015-02-28" +1 month -> "2015-03-31"
+      // On spark 3.0,     "2015-02-28" +1 month -> "2015-03-28"
```

Comment:
To Reviewer:
The case of "case AddMonths()" can be removed. But I prefer to keep it with explanation why it can't be pushdown. If the functionality for Snowflake and Spark to be the same in the future, SC can support it. This information will be good reference.

Reply:
Good.